keras_nlp/utils/preset_utils.py

```diff
@@ -100,19 +100,26 @@ def save_to_preset(
     # Include references to weights and assets.
     config["assets"] = assets
     config["weights"] = weights_filename if save_weights else None
-    recursive_pop(config, "config_config")
+    recursive_pop(config, "compile_config")
     recursive_pop(config, "build_config")
     with open(config_path, "w") as config_file:
         config_file.write(json.dumps(config, indent=4))
```

Comment:
Interesting tf bug in the logs. On keras 2 see a `json` error because of tensorflow's weird `DictWrapper` not being a real dict. We could either

- disallow calling save on keras 2 (maybe that is overall a safe idea anyway?)
- or recursively find all dict-ish things and cast them to dict

I'm inclined to do the former, what do you think? If we do that we should annotate the test with `@pytest.mark.keras_3_only`

Reply:
Yeah I agree, I think that since our saving logic is moving away from tensorflow's `DictWrapper` anyways, we should disallow calling it on keras 2. Nice catch!